src/Microsoft.Azure.Functions.Worker.Extensions.Mcp/Converters/ToolInvocationContextConverter.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Mcp.Converters;
+
+[SupportedTargetType(typeof(ToolInvocationContext))]
+internal class ToolInvocationContextConverter : IInputConverter
+{
+    public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+    {
+        ArgumentNullException.ThrowIfNull(context);
+
+        object? target = null;
```

Comment:
This shouldn't be required. It's fine to work with the typed variable created in the `TryGetToolInvocationContext` call.